### PR TITLE
Use pytest instead of deprecated nose

### PR DESCRIPTION
**Pull request by peterrus**
With the intention of letting test pass so we can work on https://github.com/thumbor-community/aws/pull/147. In theory `nose` should work with Python3 but it's a dead project and `pytest` should be plug and play in this case (tested it locally, seems fine, and the output is a lot nice imho).

@Bladrak If CircleCI is still working and you have merged https://github.com/Bladrak/docker-thumbor-dev/pull/1 we should get passing tests again by merging this into #147

Since #147 is not coming from a fork I have push access to could @amanagr maybe give me access to do so?

Alternatively, if @amanagr is not working on/interested in this project anymore maybe @Bladrak can convert #147 to a branch of this repo to be able to easier manage PR's that work on making tc_aws python3 compatible.

I think this could also just be integrated in to `master` as it should be compatible with python2. It's just that merging only this branch won't fix CI tests, we need https://github.com/Bladrak/docker-thumbor-dev/pull/1 for that as well.

### CODE REVIEW

**Comment by Bladrak** [comment]
Changed the base tu py3, once py3 is stable we can pre-release it

**Comment by Bladrak** [comment]
Also, waiting for feedback on https://github.com/Bladrak/docker-thumbor-dev/pull/1 and integrating the new image in this MR before merging it

**Comment by peterrus** [comment]
@Bladrak could it be that CircleCI is using the config from `master` instead of this branch? Because it tries to run the python2 docker image (and it should use the new python3 image).

**Comment by Bladrak** [comment]
@peterrus might be because I changed the branch of the MR afterwards and Circle is messing up

**Comment by peterrus** [comment]
Never mind @Bladrak :joy: I had a small lapse of reason. Just had to merge your upstream `py3` branch into my fork.

It seems I broke one unit test by moving to nose. I will fix that ASAP.

File: version.txt
Comment (Bladrak) [inline version.txt:1]:
```suggestion
4.0b
```

**Review by Bladrak — CHANGES_REQUESTED** [review]
Unless for the version it looks good!

**Comment by peterrus** [comment]
@Bladrak Voila!

Some funky dependency resolution going in on with regards to boto but nothing blocking! Tests pass now.

Regarding your suggestion edit in the version file: Why do we set it to 4 and not something higher than `6.2.15` ?

**Comment by Bladrak** [comment]
Sorry, you're right for the version, maybe 7.0b?

File: version.txt
Comment (Bladrak) [inline version.txt:1]:
```suggestion
7.0b
```

**Review by Bladrak — APPROVED** [review]

**Comment by Bladrak** [comment]
@peterrus looks like we have an issue on the deploy script, maybe related to the image, what do you think? https://app.circleci.com/pipelines/github/thumbor-community/aws/20/workflows/7a946629-23c4-45ff-82db-aa4e98bf4154/jobs/283

**Comment by peterrus** [comment]
Hey @Bladrak Where did you find this? I am not entirely sure what triggered this workflow.

**Comment by peterrus** [comment]
Will fix! Strange that this only manifests itself now as I saw some successful runs a few minutes back. https://app.circleci.com/pipelines/github/thumbor-community/aws/18/workflows/28a3b468-fab3-48f8-addf-d94295ee4e3e/jobs/281 for example

**Comment by Bladrak** [comment]
Yeah I think it's because the one I linked is on a branch on the repo and tries to release it, the other one might be on the PR so not trying to release it.

**Comment by peterrus** [comment]
@Bladrak ^ should fix it

**Comment by Bladrak** [comment]
@peterrus we're probably missing ssh as well

**Comment by peterrus** [comment]
Yeah ssh should be installed as a dependency of git. Let me know whats up.

**Comment by Bladrak** [comment]
The deploy key expired on the repo, I re-generated it and it seems to be fine :)

**Comment by peterrus** [comment]
Awesome! @Bladrak Will this also publish a pip package or is that only done for `master` ?

**Comment by Bladrak** [comment]
I thought so but apparently it didn't. I need to check the circle config

**Comment by peterrus** [comment]
Alright. Let me know if you need any help, happy to do so. Thanks for your quick reactions thusfar!

**Comment by Bladrak** [comment]
Released on pypi: https://pypi.org/project/tc-aws/7.0b0/
https://pypi.org/project/tc-aws/7.0b0/
Thanks for the contribution!

**Pull request merged**
